pkg/build/api/types.go

```diff
@@ -266,6 +266,18 @@ const (
 	// secret for pushing the output image.
 	// The build will stay in the pending state until the secret is created, or the build times out.
 	StatusReasonMissingPushSecret = "MissingPushSecret"
+
+	// StatusReasonFailedPostCommitHook indicates that the postcommit failed
+	StatusReasonPostCommitHookFailed = "PostCommitHookFailed"
+
+	// StatusReasonGitCloneFailed
+	StatusReasonGitCloneFailed = "GitCloneFailed"
+
+	// StatusReasonPullImageFailed
+	StatusReasonPullImageFailed = "PullImageFailed"
```

Comment:
should this be broken up into different image pulls that could have failed?

* builder image
* input image
* previous image for an incremental build?